src/mscorlib/src/System/Globalization/DateTimeFormat.cs

```diff
@@ -957,12 +949,88 @@ internal static String Format(DateTime dateTime, String format, DateTimeFormatIn
             }
 
             if (format.Length == 1) {
+                switch (format[0]) {
+                    case 'o':
```

Comment:
I will do that, it was another thing I debated and switches seemed to be used so much in this file I went with the style of the file :)